gimie/make_graph.py

```diff
@@ -0,0 +1,75 @@
+from rdflib import Graph, URIRef, parser
+import hashlib
+from calamus import fields
+from calamus.schema import JsonLDSchema
+
+schema = fields.Namespace("http://schema.org/")
+
+
+def generate_fair_uri(
+    repository_path: str = "https://example.com/Repository/",
+):
+    """given a repository_path, returns a URI with a hash for uniqueness, or the repository URL if it's online"""
+    # Compute the SHA-256 hash of the repository name
+    hash = hashlib.sha256(repository_path.encode()).hexdigest()
+    # TODO https://www.github.com/repoX should return the same result as github.com/repoX
+    if (
```

Comment:
I'm tempted to actually correct the user's mistake before validating. If they supply a url 'github.com/SDSC-ORD/gimie' I think that's valid (after all, if you put that in your browser, it works). However, it works because your browser is smart enough to correct the user error and add https:// in front of it. In that spirit, I think we can do the same sort of pre-processing:

```
if not 'https' or 'http' in URL:
   URL = https://+URL
```
   
or something along those lines.

Only after that has been done, do we run the validate_url function on it. Or is this very against conventions?